RadixWallet/Features/CardCarousel/CardCarousel+View.swift

```diff
@@ -0,0 +1,194 @@
+import Foundation
+import SwiftUI
+
+// MARK: - CardCarousel.View
+extension CardCarousel {
+	public struct View: SwiftUI.View {
+		public let store: StoreOf<CardCarousel>
+
+		private let margin: CGFloat = .medium1
+		private let spacing: CGFloat = .small1 * 0.5
+		@ScaledMetric private var height: CGFloat = 105
+		@SwiftUI.State private var selectedCardIndex = 0
+
+		public var body: some SwiftUI.View {
+			WithPerceptionTracking {
+				VStack(spacing: 8) {
+					GeometryReader { _ in
+						coreView
+					}
+					.frame(height: store.cards.isEmpty ? 0 : height)
+
+					positionIndicator
+				}
+			}
+			.onAppear {
+				store.send(.view(.didAppear))
+			}
+		}
+
+		@MainActor
+		private var coreView: some SwiftUI.View {
+			TabView(selection: $selectedCardIndex) {
+				ForEach(Array(store.cards.enumerated()), id: \.1) { index, card in
```

Comment:
could we use `\.element` instead of `\.1`?

Reply:
What is the purpose of enumerating it and then use `element` as `id`, instead of just doing `\id:.self`? And stepping back, what was wrong with having the card be `identifiable`?

Reply:
We use the index to tag each card, which is necessary for the positionIndicator defined below. We can extend the card model (now defined in Sargon), and make it `Identifiable` by defining an id for each. Same result, just need some extra code.

Reply:
The problem with that is that swiftui will not be able to handle insertion and removal animations properly if the the index is the id. For example if you have four cards and remove the third card, with id and index 2, the collection will still have a card with id 2, but it will be the one that used to have id 3. In short, using index as id with a dynamic collection is not supported in SwiftUI. That is also why I called my helper type that wraps up the functionality you use here, "ForEachStatic".

Reply:
We aren't using the `index` as the `id`, we are using the `\.element` to identify them.

We are using the `index` to tag them, and that works correctly on your example since after removing card with `index: 2`, the index (and tag) is properly updated with the new set of elements.

Reply:
Ah, alright, so you use element for id, and index only for the tag, for the benefit of the tabview. 

I would still consider using the same tag and id, and map it in the reducer, since in there you have access to both the elements and their offsets.